stub-runner/stub-runner-spring/src/test/groovy/com/ofg/stub/StubRunnerFactorySpec.groovy

```diff
@@ -0,0 +1,47 @@
+package com.ofg.stub
+import com.ofg.infrastructure.discovery.ServiceConfigurationResolver
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.test.TestingServer
+import spock.lang.Specification
+
+import java.nio.file.Files
+
+class StubRunnerFactorySpec extends Specification {
+
+    ServiceConfigurationResolver configurationResolver = Stub(ServiceConfigurationResolver)
+    TestingServer testingServer = Stub(TestingServer)
+    CuratorFramework curatorClient = Stub(CuratorFramework)
+    StubDownloader downloader = Mock(StubDownloader)
+    StubRunnerFactory factory = new StubRunnerFactory(testingServer, configurationResolver, curatorClient, downloader)
+
+    void setup() {
+        factory.withStubRepositoryRoot("http://123.0.0.1/repository")
+        testingServer.connectString >> "localhost:12345"
+        configurationResolver.dependencies >> ["pl": ["path": "com/ofg/risk-service"], eu: ["path": "com/ofg/fraud"]]
+    }
+
+    def "Should download stub definitions many times"() {
+        given:
+            File file = Files.createTempDirectory("mappings").toFile()
```

Comment:
Maybe we could use `TemporaryFolder` rule from JUnit instead (it has auto cleanup)?